hack/test-cmd.sh

```diff
@@ -690,6 +700,36 @@ runTests() {
   # Post-condition: no POD is running
   kube::test::get_object_assert 'pods --namespace=other' "{{range.items}}{{$id_field}}:{{end}}" ''
 
+  # Clean up 
+  kubectl delete namespace other
+
+  ##############
+  # Secrets #
+  ##############
+
+  ### Create a generic secret in a specific namespace
+  # Pre-condition: no SECRET exists
+  kube::test::get_object_assert 'secrets --namespace=test-secrets' "{{range.items}}{{$id_field}}:{{end}}" ''
```

Comment:
`hack-test.cmd.sh` is not running with admission control for `NamespaceLifecycle` so its actually not a prerequisite in this suite to create a `Namespace` before it is being used so there are in fact no other controllers trying to create service accounts right now.  Whether or not that is a good/bad thing is a separate topic, I think it would impact other tests in this suite outside scope of this PR.